File: pages/LoginPage.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './BasePage';

/**
 * LoginPage contains all selectors and methods related to login functionality
 */
export class LoginPage extends BasePage {
  // Selectors
  private usernameField: Locator;
  private passwordField: Locator;
  private loginButton: Locator;

  constructor(page: Page) {
    super(page);
    this.usernameField = page.getByRole('textbox', { name: 'Username' });
    this.passwordField = page.getByRole('textbox', { name: 'Password' });
    this.loginButton = page.getByRole('button', { name: 'Login' });
  }

  /**
   * Navigate to login page
   */
  async navigateToLogin() {
    await this.goto('http://digital-rx-pro.s3-website-us-east-1.amazonaws.com/');
    await this.wait(1000);
  }

  /**
   * Perform login with credentials
   */
  async login(username: string, password: string) {
    await this.fillField(this.usernameField, username);
    await this.fillField(this.passwordField, password);
    
    // Wait for navigation while clicking login
    await Promise.all([
      this.waitForURL('**/doctor/rx', 30000),
      this.clickElement(this.loginButton)
    ]);
    
    // Wait for RX page to fully load
    await this.wait(3000);
  }

  /**
   * Login with default credentials (Sadman/Sadman1#)
   */
  async loginAsDoctor() {
    await this.login('Sadman', 'Sadman1#');
  }

  /**
   * Get doctor name from top right corner
   */
  async getDoctorName(): Promise<string> {
    const doctorNameElement = this.page.getByRole('button', { name: 'User' }).first().locator('p');
    await this.waitForElement(doctorNameElement, 10000);
    return await doctorNameElement.textContent() || '';
  }

  /**
   * Check if logged in by verifying URL contains doctor/rx
   */
  isLoggedIn(): boolean {
    return this.getURL().includes('doctor/rx');
  }
}

File: pages_py/LoginPage.py
from playwright.async_api import Page
from .BasePage import BasePage


class LoginPage(BasePage):
    """LoginPage contains all selectors and methods related to login functionality"""

    def __init__(self, page: Page):
        super().__init__(page)

    async def navigate_to_login(self):
        """Navigate to login page"""
        await self.goto('http://digital-rx-pro.s3-website-us-east-1.amazonaws.com/')

    async def login(self, username: str, password: str):
        """Perform login with credentials"""
        username_field = self.page.get_by_role('textbox', name='Username')
        password_field = self.page.get_by_role('textbox', name='Password')
        login_button = self.page.get_by_role('button', name='Login')

        await self.fill_field(username_field, username)
        await self.fill_field(password_field, password)
        await self.click_element(login_button)
        await self.page.wait_for_load_state('load', timeout=60000)

    async def get_doctor_name(self) -> str:
        """Get doctor name from top right button"""
        try:
            doctor_button = self.page.locator('[class*="user"], [class*="profile"], button').filter(has_text='Dr')
            return await doctor_button.first.text_content()
        except:
            return ""

    async def is_logged_in(self) -> bool:
        """Check if user is logged in"""
        return 'doctor/rx' in self.page.url or 'doctor/patient' in self.page.url
